server/sonar-server/src/main/java/org/sonar/server/component/DefaultRubyComponentService.java

```diff
@@ -22,86 +22,55 @@
 import org.sonar.db.component.ResourceDao;
 import org.sonar.db.component.ResourceDto;
-import org.sonar.server.favorite.FavoriteUpdater;
 import org.sonar.server.organization.DefaultOrganizationProvider;
-import org.sonar.server.permission.PermissionTemplateService;
+import org.sonar.server.user.UserSession;
 import org.sonar.server.util.RubyUtils;
 
 import static org.sonar.server.component.NewComponent.newComponentBuilder;
 
-public class DefaultRubyComponentService implements RubyComponentService {
+/**
+ * Used in GOV
+ */
+public class DefaultRubyComponentService {
 
+  private final UserSession userSession;
   private final DbClient dbClient;
   private final ResourceDao resourceDao;
-  private final ComponentService componentService;
-  private final PermissionTemplateService permissionTemplateService;
-  private final FavoriteUpdater favoriteUpdater;
+  private final ComponentUpdater componentUpdater;
   private final DefaultOrganizationProvider defaultOrganizationProvider;
 
-  public DefaultRubyComponentService(DbClient dbClient, ResourceDao resourceDao, ComponentService componentService,
-    PermissionTemplateService permissionTemplateService, FavoriteUpdater favoriteUpdater,
+  public DefaultRubyComponentService(UserSession userSession, DbClient dbClient, ComponentUpdater componentUpdater, ResourceDao resourceDao,
     DefaultOrganizationProvider defaultOrganizationProvider) {
+    this.userSession = userSession;
     this.dbClient = dbClient;
+    this.componentUpdater = componentUpdater;
     this.resourceDao = resourceDao;
-    this.componentService = componentService;
-    this.permissionTemplateService = permissionTemplateService;
-    this.favoriteUpdater = favoriteUpdater;
     this.defaultOrganizationProvider = defaultOrganizationProvider;
   }
 
-  @Override
-  @CheckForNull
-  public Component findByKey(String key) {
-    try (DbSession dbSession = dbClient.openSession(false)) {
-      return dbClient.componentDao().selectByKey(dbSession, key).orNull();
-    }
-  }
-
   // Used in GOV
-  @CheckForNull
+  /**
+   * @deprecated Use {@link ComponentUpdater#create(DbSession, NewComponent, Long)} instead
+   */
+  @Deprecated
   public Long createComponent(String key, String name, String qualifier) {
-    return createComponent(key, null, name, qualifier);
-  }
-
-  // Used in rails
-  @CheckForNull
-  public Long createComponent(String key, @Nullable String branch, String name, @Nullable String qualifier) {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      return createComponent(dbSession, key, branch, name, qualifier);
+      return componentUpdater.create(
+        dbSession,
+        newComponentBuilder()
+          .setOrganizationUuid(defaultOrganizationProvider.get().getUuid())
+          .setKey(key)
+          .setName(name)
+          .setQualifier(qualifier)
+          .build(),
+        userSession.isLoggedIn() ? userSession.getUserId().longValue() : null).getId();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'getUserId' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)